File: BinarySearch/binary_search.py

def binary_search(A, x):
    n = len(A)
    if len(A) == 1:
        return A[0] == x
    
    mid = n//2
    if x > A[mid]:
        return binary_search(A[mid+1:], x)
    elif x < A[mid]:
        return binary_search(A[:mid], x)
    else:
        return A[mid] == x

    
A = [9,10,13,15,19,21,33,41]
print(binary_search(A, 55))

File: BinarySearch/binarySearch.js
const binarySearch = (A, x) => {
    const n = A.length
    if (n == 1) {
        return A[0] == x
    }

    let mid = Math.floor(n / 2)
    if (A[mid] > x) {
        return binarySearch(A.slice(0,mid), x)
    } else if (A[mid] < x) {
        return binarySearch(A.slice(mid+1, n), x)
    } else {
        return A[mid] == x
    }
}

const A = [9, 10, 13, 15, 19, 21, 33, 41]
console.log(binarySearch(A, 33))